proto/client/gClient.go

```diff
@@ -156,64 +158,60 @@ func (confClient *ConfigClient) GetConfigClientConn() *grpc.ClientConn {
-					logger.GrpcLog.Errorf("failed to subscribe: %v", err)
-					time.Sleep(time.Second * 5)
-					// Retry on failure
-					continue
-				}
-			} else if status == connectivity.Idle {
-				logger.GrpcLog.Errorln("connectivity status idle, trying to connect again")
-				time.Sleep(time.Second * 5)
-				continue
-			} else {
-				logger.GrpcLog.Errorln("connectivity status not ready")
-				time.Sleep(time.Second * 5)
-				continue
-			}
-		}
-		rsp, err := stream.Recv()
-		if err != nil {
-			logger.GrpcLog.Errorf("failed to receive message: %v", err)
-			// Clearing the stream will force the client to resubscribe on next iteration
-			stream = nil
-			time.Sleep(time.Second * 5)
-			// Retry on failure
-			continue
+	stream = nil
+	status := confClient.Conn.GetState()
+	var err error
+	if status == connectivity.Ready {
+		logger.GrpcLog.Infoln("connectivity ready")
+		rreq := &protos.NetworkSliceRequest{RestartCounter: selfRestartCounter, ClientId: myid, MetadataRequested: confClient.MetadataRequested}
+		if stream, err = confClient.Client.NetworkSliceSubscribe(context.Background(), rreq); err != nil {
+			logger.GrpcLog.Errorf("failed to subscribe: %v", err)
+			return stream
 		}
+		return stream
+	} else if status == connectivity.Idle {
+		logger.GrpcLog.Errorf("connectivity status idle, trying to connect again")
+		return stream
+	} else {
+		logger.GrpcLog.Errorf("connectivity status not ready")
+		return stream
+	}
+}
 
-		logger.GrpcLog.Infoln("stream msg received")
-		logger.GrpcLog.Debugf("network slices %d, RC of configpod %d", len(rsp.NetworkSlice), rsp.RestartCounter)
-		if configPodRestartCounter == 0 || (configPodRestartCounter == rsp.RestartCounter) {
-			// first time connection or config update
-			configPodRestartCounter = rsp.RestartCounter
-			if len(rsp.NetworkSlice) > 0 {
-				// always carries full config copy
-				logger.GrpcLog.Infoln("first time config received", rsp)
-				commChan <- rsp
-			} else if rsp.ConfigUpdated == 1 {
-				// config delete , all slices deleted
-				logger.GrpcLog.Infoln("complete config deleted")
-				commChan <- rsp
-			}
-		} else if len(rsp.NetworkSlice) > 0 {
-			logger.GrpcLog.Errorln("config received after config pod restart")
-			// config received after config pod restart
-			configPodRestartCounter = rsp.RestartCounter
+// subscribeToConfigPod subscribing channel to ConfigPod to receive configuration
+// using stream and communication channel as inputs
+func (confClient *ConfigClient) subscribeToConfigPod(commChan chan *protos.NetworkSliceResponse, stream protos.ConfigService_NetworkSliceSubscribeClient) {
+	rsp, err := stream.Recv()
+	if err != nil {
+		logger.GrpcLog.Errorf("failed to receive message: %v", err)
+		return
+	}
+
+	logger.GrpcLog.Infoln("stream msg received")
+	logger.GrpcLog.Debugf("network slices %d, RC of config pod %d", len(rsp.NetworkSlice), rsp.RestartCounter)
+	if configPodRestartCounter == 0 || (configPodRestartCounter == rsp.RestartCounter) {
```

Comment:
Ideally pulling the config from webserver would be lot more simple approach. Current way of pushing config from webconsole is bit error prone.  This may need more changes and something can be done later.

Reply:
I am planning to open PR's in each NF to unify the implementation as described above.

Reply:
I agree that it could be a good moment to revisit the approach. Having the NF get the config from webconsole would be better in a lot of ways, but we would also have to make sure that a configuration change properly triggers a reconfiguration on the NFs.

We could keep the GRPC connection to have webconsole notify subscribers that the configuration has changed, and then the NFs can get the current up to date configuration from webconsole. We would however need to take into account that for a large deployment with many subscribers, the whole config could be large.

I think this would warrant a more in-depth redesign and it might make more sense outside of this PR that is meant to fix a current severe bug.

Reply:
This issue will not be addressed in this PR as we aim to fix a bug here.